src/java.base/share/classes/java/math/BigInteger.java

```diff
@@ -118,6 +118,42 @@
  * the full supported positive range of {@code BigInteger}.
  * The range must be at least 1 to 2<sup>500000000</sup>.
  *
+ * @apiNote
+ * <a id=algorithmicComplexity>As {@code BigInteger} values are
+ * arbitrary precision integers, the algorithmic complexity of the
+ * methods of this class varies and may be superlinear in the size of
+ * the input. For example, a method like {@link intValue()} would be
+ * expected to run in <i>O</i>(1), that is constant time, since with
+ * the current internal reprentation only a fixed-size component of
+ * the {@code BigInteger} needs to be accessed to perform the
+ * conversion to {@code int}. In contrast, a method like {@link not()}
+ * would be expected to run in <i>O</i>(<i>n</i>) time where <i>n</i>
+ * is size of the {@code BigInteger} in bits, that is, to run in time
+ * proportional to the size of the input. For multiplying two {@code
+ * BigInteger} values of size <i>n</i>, a naive multiply algorithm
+ * would run in time <i>O</i>(<i>n<sup>2</sup></i>) and theoretical
+ * results indicate a multiply algorithm for numbers using this
+ * category of representation must run in <em>at least</em>
+ * <i>O</i>(<i>n</i>&nbsp;log&nbsp;<i>n</i>). Common multiply
```

Comment:
"Common _multiply_" or "Common _multiplication_"?

Reply:
Doing some grammar research ...

In English, using a noun as an adjective is called an "attributive noun," so the "multiplication" in "multiplication algorithm" would play that role.

Strictly speaking to "adjective-ifly" a verb would be to use a participle, one way to do that anyway -- "multiplying algorithm."

In any case, "multiply algorithm" reads okay to me, but I don't mind changing it to "multiplication algorithm."

Reply:
Then it's fine as it is.